File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for s, e, k in queries:
        l = []
        for i in range(s,e+1):
            if arr[i] > k:
                l.append(arr[i])
        if l:
            answer.append(min(l))
        else:
            answer.append(-1)

    return answer

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        int count = 0;
        for (int[] query : queries) {
            List<Integer> minList = new ArrayList<>();
            for (int i = query[0]; i < query[1] + 1; i++) {
                if (arr[i] > query[2]) {
                    minList.add(arr[i]);
                }
            }
            if(minList.isEmpty()){
                answer[count] = -1;
            } else {
                answer[count] = Collections.min(minList);
            }
            count++;
        }
        return answer;
    }
}